clustering/jgroups/WFLY-8770_Integrate_NATIVE_S3_PING_discovery_protocol.adoc

```diff
@@ -0,0 +1,112 @@
+= WFLY-8770 Integrate NATIVE_S3_PING discovery protocol
+:author:            Radoslav Husar
+:email:             rhusar@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+`NATIVE_S3_PING` is a discovery protocol designed for JGroups cluster nodes running in AWS EC2.
+While there are other protocols that can be used in such an environment (`TCPPING`, `JDBC_PING`), the `NATIVE_S3_PING` is an ideal fit.
+Discovery is performed by writing and reading from an S3 bucket.
+Protocol is automatically configured from system properties.
+
+Historically all protocols were bundled within main `jgroups-<version>.jar`.
+Since the main JGroups project does not allow runtime dependencies, protocols had to include a lot of embedded unmaintainable copy-pasted and trimmed library code.
+This hinders supportability.
+Protocols needing to fetch dependencies now reside in separate repositories in the https://github.com/jgroups-extras organization.
+As such, `NATIVE_S3_PING` resides in the https://github.com/jgroups-extras/native-s3-ping repository.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/EAP7-863[EAP7-863] Integrate NATIVE_S3_PING discovery protocol
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-8770[WFLY-8770] Integrate NATIVE_S3_PING discovery protocol (upstream issue)
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [x] QE
+
+=== Affected Projects or Components
+
+* WildFly
+* JBoss EAP
+
+=== Other Interested Projects
+
+* Infinispan Server already bundles this protocol.
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
```

Comment:
I don't think this RFE is relevant to the OpenShift environment since it targets EC2.